packages/botonic-plugin-contentful/src/util/backoff.ts

```diff
@@ -0,0 +1,34 @@
+export function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+export interface Backoff {
+  backoff(): Promise<void>
+}
+
+export class ExponentialBackoff implements Backoff {
+  constructor(private startMs = 10, private times = 10) {}
+  async backoff(): Promise<void> {
+    if (this.times <= 0) {
+      throw new Error('Aborting exponential backoff')
+    }
+    await sleep(this.startMs)
+    this.startMs *= 2
+    this.times++
+  }
+}
+
+export async function repeatWithBackoff<T>(
+  func: () => Promise<T>,
+  backoff = new ExponentialBackoff(),
+  logger: (msg: string) => void = console.log
+): Promise<T> {
+  for (;;) {
```

Comment:
Why do you need the `for` here if you're returning an `await`?

Reply:
in case it fails, we want to repeat. backof will be the responsible to break with exception if it times out